### PR TITLE
Many things ...

### PR DESCRIPTION
**Pull request by DanielSmedegaardBuus**
Hello again, Mr. Fisher :)

I was laid off while doing the last batch of changes, but I got rehired this year, and I've been adding more stuff to the converter.

I don't know if you want to pull in all of these changes as they're quite large, but either way, there's at least one bug fix you probably want to pull in.

I'll be continuing to work on this as I find more missing commands :+1: 

Cheers!

### CODE REVIEW

File: JavascriptFormatter.js [hunk 1]
```diff
@@ -475,13 +472,10 @@ CallSelenium.prototype.invert = function () {
...
     log.info('Processing ' + this.message);
-    if (this.message == 'waitForPageToLoad') {
-        return '';
-    }
     var result = '';
     var adaptor = new SeleniumWebDriverAdaptor(this.rawArgs);
-    if (this.message == 'getEval')
-        adaptor.rawArgs = this.args; // getEval only args available
+    if (this.message.match(/^(getEval|runScript)/))
+        adaptor.rawArgs = this.args; // getEval only args available (Daniel: I assume this means we always want escaped stringified args here; that is, the code as converted to a string, for use with browser.safeEval(<code string>), and getEval may be used elsewhere, so we still want to have that pass forth unescaped rawArgs by default...?)
```
Comment (flyingfisher) [inline JavascriptFormatter.js:478]: This is a bug. getEval rawArgs is undefined and args has its value. Here just assign it back

File: JavascriptFormatter.js [hunk 2]
```diff
@@ -548,8 +543,18 @@ function formatCommand(command) {
                             eq = eq.invert();
                         line = waitFor(eq);
                     }
+                    else if (command.command.match(/^(getEval|runScript)/)) {
+                        call = new CallSelenium(def.name, xlateArgument(command.getParameterAt(0)), command.getParameterAt(0));
```
Comment (flyingfisher) [inline JavascriptFormatter.js:547]: the second param is equal to the first?
command.getParameterAt(0),command.getParameterAt(0)? A typo?

File: JavascriptFormatter.js [hunk 3]
```diff
@@ -913,18 +956,19 @@ function array(value) {
...
+    return this.e1.toString() + " === " + this.e2.toString();
 };
 Equals.prototype.assert = function () {
-    return "assert.equal(" + this.e1.toString() + ", " + this.e2.toString()
-        + ", 'Assertion error: Expected: " + this.e1.toString() + ", Actual: ' + "
-        + this.e2.toString() + ");";
+    var varA = getTempVarName();
+    var varB = getTempVarName();
+    return "var " + varA + " = " + this.e1.toString() + ";\n" + "var " + varB + " = " + this.e2.toString() + ";\n"
+        + "assert.equal(" + varA + ", " + varB + ", 'Assertion error: Expected: ' + " + varA + " + ', got: ' + " + varB + ");";
```
Comment (flyingfisher) [inline JavascriptFormatter.js]: the assert should tell user which string is wrong, not to tell them Expected 'var1', got: 'var2'

**Comment by flyingfisher** [comment]
Excellently!
Thank you very much.
I will merge it and test it.
I will notify you if any bug happens.

**Comment by DanielSmedegaardBuus** [comment]
There's a bug with the keyUp/Down/Press thingie so far, also I'm gonna have a look at the things you mention, thank you! Don't merge yet :D

File: JavascriptFormatter.js (hunk 2 above)
Comment (DanielSmedegaardBuus) [inline JavascriptFormatter.js:547]: Indeed :D Pretty sure there shouldn't evet be a second argument to those, though. Will correct it :)

**Comment by DanielSmedegaardBuus** [comment]
I also went straight into implementing an experimental "user command extensions" thingie, so that one can extend the converter with one's own commands ... I have some issues with executing code with wd-sync though, so it might be awhile ... will report back :)

**Comment by DanielSmedegaardBuus** [comment]
~~Okay, can I ask you a question? Do you have any idea why I can't use elements with wd-sync .execute? — I'm thinking you're more skilled than me with regards to wd-sync :D~~

Nevermind, figured it out and am submitting a pull request to fix a bug in wd-sync.

**Pull request merged**